core/sql/exp/ExpHbaseInterface.cpp

```diff
@@ -1315,32 +1192,30 @@ Lng32 ExpHbaseInterface_JNI::checkAndInsertRow(
 	  const int64_t timestamp,
           NABoolean asyncOperation)
 {
-  HTableClient_JNI *htc = client_->getHTableClient((NAHeap *)heap_, tblName.val, useTRex_, hbs_);
-  if (htc == NULL) {
-    retCode_ = HBC_ERROR_GET_HTC_EXCEPTION;
-    return HBASE_OPEN_ERROR;
-  }
-  
-  Int64 transID;
+  HTableClient_JNI *htc;
+  Int64 transID; 
+  NABoolean checkAndPut = TRUE;
+
   if (noXn)
-    transID = 0;
-  else
+    transID = 0; 
+  else 
     transID = getTransactionIDFromContext();
- 
+  retCode_ = client_->insertRow((NAHeap *)heap_, tblName.val, hbs_,
```

Comment:
checkAndInsertRow and insertRow is implemented using the same method HBaseClient_JNI::insertRow. There is a parameter checkAndPut is used to determine if checkAndPut is needed